app/controller/Map.js

```diff
@@ -75,10 +75,12 @@ Ext.define('DevCycleMobile.controller.Map', {
 	* associated groups that the user is a part of
 	*/
 	addGroup: function (groupCode, groupName) {
-		var map = Ext.getCmp('mapview').map;
+		var map;
 		//Ensure the map has been loaded
-		if (map != undefined)
-		{
+		var interval = setInterval(function() {
+		map = Ext.getCmp('mapview').map;
```

Comment:
It would be easier to read if this were indented. Usually in javascript when you define a function, everything inside it is indented one level further.